### DISCUSSION TITLE
Question 3: lack of data in competition

### DISCUSSION THREAD

**Discussion by mybirth0407** [Boostcamp-AI-Tech-1-15/ddobokki_log#31] [2. DL Basic]
Train data가 10,000개, Test data가 1,000,000인 vision competition이 있습니다. 어떤 방법으로 문제를 해결하실 건가요? (일부 공개된 Test data가 1,000,000 이라고 가정)

**Comment by ddobokki** [comment]
이번 pstage보다 극단적인 상황이군요....
추가 데이터가 사용가능하면 우선 데이터부터 찾아야 할것 같습니다.
불가능하다면 Augmentation을 정교하게 해야할것 같습니다. 
데이터 자체가 적으므로 굳이 딥러닝을 기법을 사용하지 않고 이미지의 구간을 나누어 평균이든 max pooling이든 피쳐를 적게 나누어 svm이나 randomforest를 사용해 볼 수도 있을것 같습니다.

**Reply by mybirth0407** [reply]
네 좋습니다. 극단적인 상황일수록 다양한 방법들들을 생각해볼 수 있는데요! "데이터 자체가 적으므로 굳이 딥러닝을 기법을 사용하지 않고" 보다는 "데이터 수가 적으므로 학습 기반 모델보다 통계 기반의 모델이 더 잘 작동할 것이다." 정도가 더 어울릴 것 같아요!

**Comment by minji-o-j** [comment]
test data의 분포가 train data와 비슷하다는 확신이 있고, 모든 class의 비율이 일정하다면 크게 문제가 되지 않을 것 같습니다.

하지만 대부분 competition에는 이러한 보장이 없기 때문에, 저라면 우선 10000개의 데이터에 대해 EDA를 해볼 것 같습니다. 데이터가 10000개지만 그 안에서도 data imbalance가 있을 수도 있기 때문입니다. 

만약에 data imbalance가 있다면, 적은 이미지를 가진 class에 대해 양질의 데이터를 모아서 데이터의 균형을 맞출 것 같습니다. 여기서 제가 말하는 양질의 데이터란, class간 특징이 뚜렷하게 나타나있거나 train set이랑 비슷한 조건에서 수집된 데이터를 의미합니다.
데이터 추가가 불가능하다면 데이터를 다른 class보다 더 augmentation하거나 weight를 더 주거나, 이미지의 크기가 크지 않은 경우 oversampling을 할 것 같습니다.

data imbalance를 해소한 이후에는 최대한 다양한 방식으로 augmentation할 것 같습니다. 하지만 이때 train set에 변형이나 왜곡이 심한 augmentation 기법은 사용하지 않을 것 같습니다.  왜냐하면 현재로서 test set에 어떤 data가 있는지 알 수 없기 때문에 data의 형태가 train dataset과 비슷할 것이라는 가정을 가지고 실험을 진행할 것 같다고 생각하기 때문입니다.

**Reply by mybirth0407** [reply]
좋습니다! 먼저 데이터의 분포를 체크하는 것 너무 좋습니다! EDA를 통해 insight를 얻는 점도 좋구요! augmentation의 경우는 이미지, signal과 같은 데이터는 사용할 수 있지만, table 형태의 데이터에는 사용하기 어려워서, 이런 부분에 대해서는 말씀하신대로 weight를 주는 방식도 있을 것 같네요! 좋은 답변입니다!

**Reply by minji-o-j** [reply]
@mybirth0407 signal data 신호를 증폭이 아니라 데이터 자체를 늘리는 것도 가능한가요?

**Reply by mybirth0407** [reply]
> @mybirth0407 signal data 신호를 증폭이 아니라 데이터 자체를 늘리는 것도 가능한가요?

이제 체크했는데요..! 신호들을 합성하거나, 피치 조절 등을 통해 데이터 자체를 변경하여 수를 늘리는 것도 가능해요!
